packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -388,6 +410,28 @@ class SliverChildBuilderDelegate extends SliverChildDelegate {
   /// Defaults to providing an index for each widget.
   final SemanticIndexCallback semanticIndexCallback;
 
+  /// Called to find the new index of a child based on its key in case of reordering.
+  ///
+  /// If not provided, a child widget may not map to its existing [RenderObject]
+  /// when the order in which children are returned from [builder] changes.
+  /// This may result in state-loss.
+  final Function findChildIndex;
```

Comment:
never use `Function`, always use a typedef.

If this is a callback, it shouldn't be called `somethingIndex`. Something called `somethingIndex` should be of type `int`.

Maybe `childIndexGetter` or `findChildIndexCallback` or some such.

Reply:
fixed